tpm2/structures.go

```diff
@@ -107,8 +107,8 @@ func (h TPMHandle) HandleValue() uint32 {
 // only PCR, session, and permanent values have known constant Names.
 // See definition in part 1: Architecture, section 16.
 func (h TPMHandle) KnownName() *TPM2BName {
-	switch (byte)(h >> 24) {
-	case 0x00, 0x02, 0x03, 0x40:
+	switch (TPMHT)(h >> 24) {
+	case TPMHTPCR, TPMHTHMACSession, TPMHTPolicySession, TPMHTPermanent, TPMHTTransient:
```

Comment:
Are you sure it's right to add `TPMHTTransient` here? A transient object's name is not just the big-endian byte array of its handle, unlike with PCR, HMAC and Policy sessions, and Permanent entities.

(Thank you for the cleanup though, putting the actual symbols here!)

Reply:
Thanks for calling this out, I mis-interpreted Part 2 Structures section 14.6.2 _savedHandle_.

Based on the table from Part Architecture section 16 Table 3,  I changed the handling in https://github.com/google/go-tpm/pull/351/commits/0d435558a7e4bfdd75116dece576f6601fd1f1df to address the case of a sequence object and to allow for Name computation of Transient and Persistent Objects in future implementations.

I first thought of changing the  `AuthHandle{}` `KnownName()` method because passing in an empty `TPM2BName.Name`<sup>1</sup> would default to `TPM2BName.Handle.KnownName()` being called, which will always be `nil` because the _Name_ associated with a _sequenceHandle_ is an empty buffer<sup>1</sup>.  The `nil` _Name_ value would then error in `cmdNames`.

Rather than modify the `AuthHandle` struct method I opted to extend the _Name_ computation since it is defined for both Transient and Persistent Objects in Part Architecture section 16 Table 3.

Concerns or suggestions?

<sup>1</sup> Part 1: Architecture, section 32.4.5